samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/evaluation/TotalScoresBean.java

```diff
@@ -863,11 +865,36 @@ public List getSectionFilterSelectItems() {
 	    filterSelectItems.add(new SelectItem(TotalScoresBean.ALL_SECTIONS_SELECT_VALUE, ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.EvaluationMessages", "all_sections")));
 	    // TODO If there are unassigned students and the current user is allowed to see them, add them next.
 
-	    // Add the available sections.
-	    for (int i = 0; i < availableSections.size(); i++) {
-	        CourseSection section = (CourseSection)availableSections.get(i);
-	        filterSelectItems.add(new SelectItem(String.valueOf(i), section.getTitle()));
-	        //filterSelectItems.add(new SelectItem(section.getUuid(), section.getTitle()));
+	    String userId = AgentFacade.getAnonymousId();
+	    try {
+	        Site site = siteService.getSite(toolManager.getCurrentPlacement().getContext());
+	        GradingSectionAwareServiceAPI service = new GradingSectionAwareServiceImpl();
+	        // Add the available sections to which it belongs
+	        Collection<Group> groups = site.getGroups();
+	        int i = 0;
+	        boolean addedMoreFilterSelect = false;
+	        for (CourseSection section : availableSections) {
+	            String uuid = ((CourseSection) section).getUuid();
+	            for (Group group : groups) {
+	                if (uuid.contains(group.getId())) {
+	                    if(group.getMember(userId)!=null) {
+	                        filterSelectItems.add(new SelectItem(String.valueOf(i), group.getTitle()));
+	                        addedMoreFilterSelect = true;
+	                    }
+	                    i++;
```

Comment:
is the incrementing supposed to happen here or inside the IF statement above?

Reply:
It's OK here